pkg/storage/cache/dynamodb.go

```diff
@@ -198,24 +199,28 @@ func (d *DynamoDBDriver) HasBlob(digest godigest.Digest, path string) bool {
 func (d *DynamoDBDriver) DeleteBlob(digest godigest.Digest, path string) error {
 	marshaledKey, _ := attributevalue.MarshalMap(map[string]interface{}{"Digest": digest.String()})
 
-	expression := "DELETE BlobPath :i"
+	expression := "DELETE DuplicateBlobPath :i"
 	attrPath := types.AttributeValueMemberSS{Value: []string{path}}
 
-	_, err := d.client.UpdateItem(context.TODO(), &dynamodb.UpdateItemInput{
-		Key:                       marshaledKey,
-		TableName:                 &d.tableName,
-		UpdateExpression:          &expression,
-		ExpressionAttributeValues: map[string]types.AttributeValue{":i": &attrPath},
-	})
-	if err != nil {
+	if err := d.updateItem(digest, expression, map[string]types.AttributeValue{":i": &attrPath}); err != nil {
 		d.log.Error().Err(err).Str("digest", digest.String()).Str("path", path).Msg("unable to delete")
 
 		return err
 	}
 
-	result, _ := d.GetBlob(digest)
+	originBlob, _ := d.GetBlob(digest)
+	// if original blob is the one deleted
+	if originBlob == path {
+		// move duplicate blob to original, storage will move content here
+		originBlob, _ = d.getDuplicateBlob(digest)
```

Comment:
My understanding is the original blob is also found in the duplicate list.
Seems like this method always returns the first blob in the duplicates list.
Is there a chance the original blob is the first one in the duplicate list?

Reply:
no, there's no chance because I already removed it from the duplicate list earlier, and in duplicate list there can not be the same path multiple times, because it's a set with unique strings. right?

Reply:
I don't know if the are unique. Are you sure?

Reply:
yes, https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.NamingRulesDataTypes.html

I'll test it.